PWGHF/HFL/TableProducer/electronSelectionWithTpcEmcal.cxx

```diff
@@ -211,6 +233,45 @@ struct HfElectronSelectionWithTpcEmcal {
     }
     return true;
   }
+  // Associated electron Selection Cut
+  template <typename T>
+  bool selAssoTracks(T const& track)
+  {
+    if (!track.isGlobalTrackWoDCA()) {
+      return false;
+    }
+    if (std::abs(track.dcaXY()) > dcaXYTrackMax || std::abs(track.dcaZ()) > dcaZTrackMax) {
+      return false;
+    }
+    if (track.eta() < etaAssoTrackMin || track.eta() > etaAssoTrackMax) {
+      return false;
+    }
+
+    if (track.pt() < ptAssoTrackMin) {
+      return false;
+    }
+    if (track.tpcNSigmaEl() < tpcNsigmaAssoElectronMin || track.tpcNSigmaEl() > tpcNsigmaAssoElectronMax) {
+      return false;
+    }
+
+    return true;
+  }
+
+  // mc gen particle  selection cut
+  template <typename T>
+  bool mcGensel(T const& track)
```

Comment:
I didn't understand the need this function

Reply:
This function contains the kinematic cuts for mcgen particles. Similarly, for mc data and mc rec, I have created separate functions for track selection. Following the same approach, I have created this function for mcgen as well.

Reply:
If I didn't overlook, you are using the same kinematic variables for Mc reco, data, and Mc gen. Then why do you need different function?

Reply:
yes, We are using the same kinematic variables for MC reco, data, and MC gen, but MC reco and data have additional track selection cuts, so we need a different function.